pkg/controller/clustersync/clustersync_controller_test.go

```diff
@@ -2419,6 +2421,186 @@ func TestReconcileClusterSync_UpsertToSyncResourceApplyMode(t *testing.T) {
 	}
 }
 
+func TestReconcileClusterSync_WithParameters(t *testing.T) {
+	syncsetString := `---
```

Comment:
Funny, I thought it was a lot easier to read this way than built line by line as a go object. However, the reason is as (I think) you surmised below: I was finding it difficult to supply the `resources` -- which need to be some form of RawExtension -- via the builder.

But... that was when I still thought I might be able to get away with templates for non-string fields. Given the limitation I have now made great pains to describe, it's probably not necessary, and I can probably use the builder (and existing apply matcher).

(That said, the existing apply matcher is really difficult to understand when it fails. I may end up tweaking it.)

I'll play with it tomorrow.

Reply:
Okay, yah, since I'm no longer concerned with trying to wedge templates into non-string types, it turned out to be completely possible to use the builder. Though I still prefer stuffing in the resource as a yaml string; and I kept the apply matcher cause it really does make the results easier to read.

Reply:
Maybe this is a preference thing, but I find it extremely easy to read the Builder pattern for tests, as it is easy to pay attention to the necessary changes you want for the CR. Plus, now that you have taken the pains to understand how yaml resources need to be applied and matched, if I need to ever use it for my tests, I wouldn't need to understand how it all works - as long as I give it a string that can be converted to JSON, it'll all just work.